lib/email_log_interceptor.rb

```diff
@@ -0,0 +1,14 @@
+class EmailLogInterceptor
+  def self.delivering_email(mail)
+    Email.create!(
+      to: mail.to.first,
+      subject: mail.subject,
+      body: mail.body.encoded,
+      notify_reference: mail.header['reference']&.value,
+    )
```

Comment:
Ah, now that I've read this, I'm not sure if it's actually trivial to infer the `application_form` from the `mail` object that we get here? Depending on the recipient it could be tricky

Reply:
Yeah, it would be tricky. Adding a reference everywhere at some point might be worth it though.